packages/react-dom/src/__tests__/ReactDOMInput-test.js

```diff
@@ -716,6 +716,32 @@ describe('ReactDOMInput', () => {
     ).toBe(true);
   });
 
+  it('should not set a value for submit buttons when undefined is supplied', () => {
+    let stub = <input type="submit" value={undefined} />;
+    stub = ReactTestUtils.renderIntoDocument(stub);
+    const node = ReactDOM.findDOMNode(stub);
+
+    expect(
+      !node.hasAttribute('value') || node.getAttribute('value').length > 0,
```

Comment:
This is confusing. Which one of the two do you expect in jsdom environment? Please be specific. We don't run these tests in browsers so you don't need to account for different behavior.

Reply:
Heh, good spot. This is a bit of copy-pasta from the tests above. Will fix.